flag.go

```diff
@@ -419,20 +419,26 @@ func (f *FlagSet) PrintDefaults() {
 	fmt.Fprintf(f.out(), "%s", usages)
 }
 
-// isZeroValue guesses whether the string represents the zero
-// value for a flag. It is not accurate but in practice works OK.
-func isZeroValue(value string) bool {
-	switch value {
-	case "false":
-		return true
-	case "<nil>":
-		return true
-	case "":
-		return true
-	case "0":
+// defaultIsZeroValue returns true is the default value for this flag represents
+// a zero value.
+func (f *Flag) defaultIsZeroValue() bool {
```

Comment:
Feel free to play around with that idea, but I haven't been able to make it work yet.